pkg/controller/migcluster/migcluster_controller.go

```
@@ -123,64 +124,103 @@ func (r *ReconcileMigCluster) Reconcile(request reconcile.Request) (reconcile.Re
 		}
 	}
-	if !migCluster.Status.IsReady() {
-		return reconcile.Result{}, nil
+
+	// Remote Watch.
+	if !cluster.Status.HasBlockerCondition() {
+		err = r.setupRemoteWatch(cluster)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Ready
+	cluster.Status.SetReady(
+		!cluster.Status.HasBlockerCondition(),
+		ReadyMessage)
+
+	// End staging conditions.
+	cluster.Status.EndStagingConditions()
+
+	// Apply changes.
+	err = r.Update(context.TODO(), cluster)
+	if err != nil {
+		return reconcile.Result{}, err
 	}
 
-	// Create a Remote Watch for this MigCluster if one doesn't exist
+	// Done
+	return reconcile.Result{}, nil
+}
+
+// Update an annotation to force the `resourceVersion` to be incremented.
+// This needed to ensure a reconcile gets triggered for any resource watching
+// this resource for remote watch event propagation.
+func (r *ReconcileMigCluster) updateAnnotation(cluster *migapi.MigCluster) {
+	if cluster.Annotations == nil {
+		cluster.Annotations = make(map[string]string)
+	}
+	cluster.Annotations["reconcile-id"] = uuid.New().String()
+}
+
+// Setup remote watch.
+func (r *ReconcileMigCluster) setupRemoteWatch(cluster *migapi.MigCluster) error {
+	nsName := types.NamespacedName{
+		Namespace: cluster.Namespace,
+		Name:      cluster.Name,
+	}
 	remoteWatchMap := remote.GetWatchMap()
-	remoteWatchCluster := remoteWatchMap.Get(request.NamespacedName)
+	remoteWatchCluster := remoteWatchMap.Get(nsName)
+	if remoteWatchCluster != nil {
+		return nil
+	}
 
-	if remoteWatchCluster == nil {
-		log.Info("Starting remote watch.", "cluster", request.Name)
+	log.Info("Starting remote watch.", "cluster", cluster.Name)
 
-		var restCfg *rest.Config
-		if migCluster.Spec.IsHostCluster {
-			restCfg, err = config.GetConfig()
-			if err != nil {
-				return reconcile.Result{}, err
-			}
-		} else {
-			restCfg, err = migCluster.BuildRestConfig(r.Client)
-			if err != nil {
-				return reconcile.Result{}, nil // don't requeue
```

Did we mean to swallow this error?